src/main/java/net/ripe/ipresource/IpAddress.java

```diff
@@ -41,13 +41,17 @@ public static IpAddress parse(String s) {
 
     public static IpAddress parse(String s, boolean defaultMissingOctets) {
         try {
-            try {
-                return Ipv4Address.parse(s, defaultMissingOctets);
-            } catch (IllegalArgumentException e) {
+            if (findFirstDotOrColumn(s) == ':') {
                 return Ipv6Address.parse(s);
+            } else {
+                try {
```

Comment:
Would it makes sense to check for findFirstDotOrColumn(s) == Character.MIN_VALUE here? Or just let it throw an exception?

Reply:
"Am I right thinking that there is no IPv6 notation that allows dots before colons?"

I believe so, this one https://tools.ietf.org/html/rfc5952#page-11 says that dots never precede colons.

Reply:
Yeah, thanks, I pushed an update